app/controllers/archived_conversations_controller.rb

```diff
@@ -0,0 +1,37 @@
+class ArchivedConversationsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @conversations = load_archived
+  end
+
+  def update
+    @conversation ||= Conversation.find(params[:id])
+
+    # update the unarchived conversation to archive of a user type
+    if @conversation.unarchived_by?(current_user)
+      @conversation.archive_from(user_type)
+      load_all
+      redirect_to conversations_path
+    else
+      @conversation.update("#{user_type}_archived_at": nil)
+      @conversations = load_archived
+      redirect_to archived_conversations_path
+    end
+  end
```

Comment:
If we instead use POST/DELETE routes, similar to `ReadNotificationsController`, we can clean up even more of this logic.

Reply:
`BlocksController` might also offer some tips for how to move some logic to the model.

Reply:
Done ✅ 

But the `create` action doesn't work in a RESTFUL style. With routes set as this:
```
    resources :archived_conversations, only: %i[index create destroy], path: "/conversations/archived"
```
when making a `post` request, it shows this error:
```
Started POST "/1/conversations/archived"
ActionController::RoutingError (No route matches [POST] "/1/conversations/archived"):
```
So I have created the `create` route manually:
```
    post "conversations/archived/:id", to: "archived_conversations#create"
```
And this works fine!

Reply:
I think we need two routes. One nested under conversations for CREATE and DELETE. And another to view the list at `/conversations/archived`.

Reply:
The `delete` route works perfectly fine but not `create`. 

> One nested under conversations for CREATE 

Do you want me to move the `create` logic in `ConversationsController`? I don't think so & if that's the case then I haven't understood properly!

Reply:
Create should live under Archived not base Conversations. What isn’t working?

Reply:
How's this for a RESTful Resource?
```
    resources :archived_conversations, only: %i[index destroy], path: "/conversations/archived"
    resources :archived_conversations, only: %i[create], path: "/conversations/archived/:id"
```